Clarify tracestate header size

mwear opened the pull request:
This addresses issue #310.

Previously, the spec listed tracestate limits in two places, and used the phrasing at most 512 bytes, in one sentence, and at least 512 characters in another. The intent that is that tracing systems should propagate 512 characters, but can choose to propagate more or less depending on their situation. This also makes the change that white space characters  are included in the 512 character recommendation.

The two sentences have been consolidated and clarified under the tracestate limits section.

If there are any suggestions on wording, I'd be happy to address them.

yurishkuro commented on spec/20-http_header_format.md:
```diff
@@ -394,9 +391,9 @@ Platforms and tracing systems MUST propagate this field. There might be multiple
-Platforms and tracing vendors SHOULD propagate at least 512 characters of a
-combined header. This length includes commas required to separate list items.
-But does not include optional white space (`OWA`) characters.
+Platforms and tracing vendors _SHOULD_ propagate at least 512 characters of a
```
we don't use italic for MUST/SHOULD words, capitalization is enough.

mwear replied:
We decided that we will not be using italics going foward, see [this comment](https://github.com/w3c/trace-context/pull/307#issuecomment-511920186), but will address removal in a future PR.

yurishkuro reviewed (commented):
This PR is a small semantic change to the spec. Do we keep a  changelog?

mtwo reviewed (approved):
Approved pending resolution of Yuri's comment

mwear committed:
Clarify tracestate header size

Previously, the spec listed tracestate limits in two places, and
used the phrasing at most 512 bytes, in one sentence, and at least
512 characters in another. The intent that is that tracing systems
should propagate 512 characters, but can choose to propagate more
or less depending on their situation.

This commit consolidates and clarifies those comments under the
tracestate limits section.
